### PR TITLE
[PPD-248] state management

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- added status check for ec and service

#### Motivation and Context

- changed the behavior in case of organization or service disabled

#### How Has This Been Tested?

- junit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/spontaneouspayment/service/PaymentsService.java [hunk 1]
```diff
@@ -70,28 +71,48 @@ public class PaymentsService {
...
         return createDebtPosition(organizationFiscalCode, orgConfiguration, serviceConfiguration, spontaneousPayment);
     }
+    
+    /**
+     * check if the credit institution's status is ENABLED
+     *
+     * @param orgConfiguration   the Organization entity to check
+     * @throws AppException if the status is not suitable for the creation of the debt position
+     */
+    private void checkOrganizationStatus(Organization orgConfiguration) {
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/spontaneouspayment/service/PaymentsService.java:96]: change it in `isOrganizationEnable`

File: src/main/java/it/gov/pagopa/spontaneouspayment/controller/IEnrollmentsController.java [hunk 2]
```diff
@@ -40,7 +40,9 @@ public interface IEnrollmentsController {
             @ApiResponse(responseCode = "201", description = "Request created.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(name = "OrganizationModelResponse", implementation = OrganizationModelResponse.class))),
             @ApiResponse(responseCode = "400", description = "Malformed request.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class))),
             @ApiResponse(responseCode = "401", description = "Wrong or missing function key.", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "404", description = "Not found the service to enroll.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class))),
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/spontaneouspayment/controller/IEnrollmentsController.java]: right, we missed this scenario

File: src/main/java/it/gov/pagopa/spontaneouspayment/exception/AppError.java [hunk 3]
```diff
@@ -8,8 +8,10 @@
 public enum AppError {
 
     ORGANIZATION_NOT_FOUND(HttpStatus.NOT_FOUND, "Organization not found", "Not found the Organization Fiscal Code %s"),
+    ORGANIZATION_NOT_ENABLED(HttpStatus.UNPROCESSABLE_ENTITY, "Organization not enabled", "The organization is not in suitable state [Organization Fiscal Code = %s; Status = %s]"),
     ORGANIZATION_SERVICE_NOT_FOUND(HttpStatus.NOT_FOUND, "Not found a relation between organization and service", "Not found a relation between Organization Fiscal Code %s and Service Id %s"),
     SERVICE_NOT_FOUND(HttpStatus.NOT_FOUND, "The service configuration was not found", "Not found a service configuration for Service Id %s"),
+    SERVICE_NOT_ENABLED(HttpStatus.UNPROCESSABLE_ENTITY, "Service not enabled", "The service is not in suitable state [Service Id = %s; Status = %s]"),
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/spontaneouspayment/exception/AppError.java:14]: `Status = %s` it's not useful, we already  have `"Service not enabled"`

File: src/main/java/it/gov/pagopa/spontaneouspayment/exception/AppError.java [hunk 4]
```diff
@@ -8,8 +8,10 @@
 public enum AppError {
 
     ORGANIZATION_NOT_FOUND(HttpStatus.NOT_FOUND, "Organization not found", "Not found the Organization Fiscal Code %s"),
+    ORGANIZATION_NOT_ENABLED(HttpStatus.UNPROCESSABLE_ENTITY, "Organization not enabled", "The organization is not in suitable state [Organization Fiscal Code = %s; Status = %s]"),
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/spontaneouspayment/exception/AppError.java:11]: Status = %s it's not useful, we already have "Organization not enabled"

**Review by pasqualespica — CHANGES_REQUESTED** [review]
LGTM 👍

File: src/main/java/it/gov/pagopa/spontaneouspayment/exception/AppError.java (hunk 3 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/spontaneouspayment/exception/AppError.java:14]: Right reporting. Corrected

File: src/main/java/it/gov/pagopa/spontaneouspayment/exception/AppError.java (hunk 4 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/spontaneouspayment/exception/AppError.java:11]: Right reporting. Corrected

**Review by pasqualespica — APPROVED** [review]
👍

**Pull request merged**